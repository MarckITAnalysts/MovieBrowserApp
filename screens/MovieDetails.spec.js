import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetailScreen from './MovieDetailScreen';

jest.useFakeTimers()

jest.mock('react-native-snap-carousel', () => 'Carousel');
jest.mock('expo-status-bar', () => ({
  ...jest.requireActual('expo-status-bar'),
  StatusBar: () => {},
}));
jest.mock('react-native-progress', () => ({
  ...jest.requireActual('react-native-progress'),
  Progress: () => {},
}));

const mockedUsedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockedUsedNavigate,
  useRoute: () => ({params: {item: {id: 1}}})
}));

jest.mock('React', () => ({
  ...jest.requireActual('React'),
  useEffect: jest.fn()
}));


describe('Movie Detail Screen test cases', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MovieDetailScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
