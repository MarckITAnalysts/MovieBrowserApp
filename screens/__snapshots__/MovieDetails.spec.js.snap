// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail Screen test cases renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "paddingBottom": 20,
    }
  }
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#171717",
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "width": "100%",
          },
        ]
      }
    >
      <RNCSafeAreaView
        edges={
          {
            "bottom": "additive",
            "left": "additive",
            "right": "additive",
            "top": "additive",
          }
        }
        style={
          [
            {
              "position": "absolute",
            },
            {
              "zIndex": 20,
            },
            {
              "width": "100%",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingLeft": 16,
              "paddingRight": 16,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#eab308",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "opacity": 1,
              "paddingBottom": 4,
              "paddingLeft": 4,
              "paddingRight": 4,
              "paddingTop": 4,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            aria-hidden="true"
            bbHeight="28"
            bbWidth="28"
            color="white"
            fill="none"
            focusable={false}
            height="28"
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeWidth={2.5}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 28,
                  "width": 28,
                },
              ]
            }
            tintColor="white"
            vbHeight={24}
            vbWidth={24}
            width="28"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeWidth="2.5"
            >
              <RNSVGPath
                d="M15.75 19.5L8.25 12l7.5-7.5"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            aria-hidden="true"
            bbHeight="35"
            bbWidth="35"
            color="white"
            fill="currentColor"
            focusable={false}
            height="35"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 35,
                  "width": 35,
                },
              ]
            }
            tintColor="white"
            vbHeight={24}
            vbWidth={24}
            width="35"
          >
            <RNSVGGroup
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </RNCSafeAreaView>
      <View>
        <Image
          source={
            {
              "uri": "https://img.myloview.com/stickers/white-laptop-screen-with-hd-video-technology-icon-isolated-on-grey-background-abstract-circle-random-dots-vector-illustration-400-176057922.jpg",
            }
          }
          style={
            {
              "height": 733.7,
              "width": 750,
            }
          }
        />
        <ViewManagerAdapter_ExpoLinearGradient
          colors={
            [
              0,
              3424065303,
              4279703319,
            ]
          }
          endPoint={
            [
              0.5,
              1,
            ]
          }
          startPoint={
            [
              0.5,
              0,
            ]
          }
          style={
            [
              {
                "position": "absolute",
              },
              {
                "bottom": 0,
              },
              {
                "height": 533.6,
                "width": 750,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": -120.06,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
            },
            {
              "textAlign": "center",
            },
            {
              "fontSize": 30,
              "lineHeight": 36,
            },
            {
              "fontWeight": "700",
            },
            {
              "letterSpacing": 1,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "marginLeft": 16,
              "marginRight": 16,
            },
            [
              {
                "marginTop": 12,
              },
            ],
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#a3a3a3",
            },
            {
              "marginLeft": 16,
              "marginRight": 16,
            },
            {
              "letterSpacing": 0.25,
            },
            [
              {
                "marginTop": 12,
              },
            ],
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
